### DISCUSSION TITLE
rpmBuild.spec 一瞥

### DISCUSSION THREAD

**Discussion by xxxxxliil** [gojue/ecapture#643] [General]
https://github.com/gojue/ecapture/blob/d50ee780c0c26df50bdcd262e792b02cef55b1f4/builder/rpmBuild.spec#L12-L15
没有定义 bpftool 作为依赖项，我也没有检查目前是否只需要这些依赖即可编译与运行

https://github.com/gojue/ecapture/blob/d50ee780c0c26df50bdcd262e792b02cef55b1f4/builder/rpmBuild.spec#L5
应 https://github.com/gojue/ecapture/issues/507 期望，把 AGPL 改为 Apache 2。这里可能会有隐藏的法律问题：在之前的贡献者是否放弃了贡献的所有权、他们是否愿意自己提交的贡献以 Apache 2 协议公开。但是实践中似乎贡献完了都不太在乎

最后是注意到了 github 在单击任意行左侧，点击 ··· 呼出菜单，再点击 [Reference in new discussion](https://github.com/gojue/ecapture/discussions/new?permalink=https%3A%2F%2Fgithub.com%2Fgojue%2Fecapture%2Fblob%2Fd50ee780c0c26df50bdcd262e792b02cef55b1f4%2Fbuilder%2FrpmBuild.spec%23L11)，可以注意到大大的 `Sorry, we didn't recognize that category! Please choose one of the following valid categories.` 选择任意 Categories，body 中完全没有对应的连接

**Comment by cfc4n** [comment]
> 没有定义 bpftool 作为依赖项，我也没有检查目前是否只需要这些依赖即可编译与运行

实际上，ecapture的deb包是不依赖任何文件的，是一个GO/CGO静态编译的可执行程序。 我后面把这几个`BuildRequires`去掉。

**Reply by xxxxxliil** [reply]
从源编译的话还是有必要的，而且这是 rpm 打包用的 spec 文件，我有时间完善一下吧

**Reply by cfc4n** [reply]
编译不依赖spec文件，是可以去掉的。

如果有时间做贡献的话，https://github.com/orgs/gojue/projects/1/views/1 这里有很多TODO项，可以选一些任务项，感谢。

**Reply by xxxxxliil** [reply]
> 编译不依赖spec文件，是可以去掉的。

在 rpm 发行版打包时会用到……不过目前为止看起来似乎没有 fedora 与 openSusE 用户尝试打包

> 如果有时间做贡献的话，https://github.com/orgs/gojue/projects/1/views/1 这里有很多TODO项，可以选一些任务项，感谢。

好的

**Comment by cfc4n** [comment]
> 最后是注意到了 github 在单击任意行左侧，点击 ··· 呼出菜单，再点击 [Reference in new discussion](https://github.com/gojue/ecapture/discussions/new?permalink=https%3A%2F%2Fgithub.com%2Fgojue%2Fecapture%2Fblob%2Fd50ee780c0c26df50bdcd262e792b02cef55b1f4%2Fbuilder%2FrpmBuild.spec%23L11)，可以注意到大大的 Sorry, we didn't recognize that category! Please choose one of the following valid categories. 选择任意 Categories，body 中完全没有对应的连接

这个问题，好像是github discussion的问题？

**Reply by xxxxxliil** [reply]
先写在这，看什么时候有人说这个问题已经被 M$ 修好了
